### PR TITLE
Create Dunning Candidates

### PR DESCRIPTION
**Pull request by TheBestPessimist**
https://github.com/metasfresh/metasfresh-e2e/issues/129

### CODE REVIEW

File: cypress/integration/dunning/create_dunning_candidates.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  //   filterBySalesInvoiceNumber(siDocumentNumber);
+  //
+  //   DunningCandidates.getRows().should('have.length', 0);
+  // });
+
+
+  it('Create Dunning Candidates', function () {
+    DunningCandidates.visit();
+    cy.wait(1000); // without this wait, the action menu is not properly loaded
+    cy.wait(10000); // maybe this fixes some cache invalidation problem that Dunning type is not found in `de.metas.dunning.api.impl.DunningDAO.retrieveDunnings` (eventually, we should get rid of this sleep)
```
Comment (metas-ts) [inline cypress/integration/dunning/create_dunning_candidates.js]: i think with the fixes in metasfresh we don't need these 11sec wait any more..right?

File: cypress/integration/dunning/create_dunning_candidates.js [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+      siDueDate = dueDate;
+    });
+
+    cy.getSalesInvoiceTotalAmount().then(amount => {
+      siTotalAmount = amount;
+    });
+  });
+
+
+  // it('Ensure there are no Dunning Candidates', function () {
```
Comment (metas-ts) [inline cypress/integration/dunning/create_dunning_candidates.js]: pls clean this up. on real-live-instances there might always be dunning candidates

File: cypress/integration/dunning/create_dunning_candidates.js [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+      bpartner.apply();
+    });
+  });
+
+  it('Prepare sales invoice', function () {
+    cy.fixture('sales/sales_invoice.json').then((salesInvoiceJson) => {
+      new SalesInvoice(businessPartnerName, salesInvoiceTargetDocumentType)
+        .addLine(
+          new SalesInvoiceLine().setProduct(productName).setQuantity(originalQuantity)
+          // todo @dh: how to add a "per test" packing item
```
Comment (metas-ts) [inline cypress/integration/dunning/create_dunning_candidates.js]: i think these commented out lines are not useful in the spec; 
I.e. i don't think we plan to test dunning particularly on packing materials in the future

File: cypress/support/commands/action.js [hunk 4]
```diff
@@ -1,4 +1,15 @@
 function executeHeaderAction(actionName) {
+  /*
+  if this is run straight after visiting a window, the action is not loaded, and everything gets broken.
+  example:
+
+    cy.visitWindow('540154);
```
Comment (metas-ts) [inline cypress/support/commands/action.js]: i think these 3 lines don'T add value and can go; correct?

File: cypress/integration/dunning/create_dunning_candidates.js (hunk 1 above)
Comment (TheBestPessimist) [reply cypress/integration/dunning/create_dunning_candidates.js]: removed the sleep

File: cypress/integration/dunning/create_dunning_candidates.js (hunk 2 above)
Comment (TheBestPessimist) [reply cypress/integration/dunning/create_dunning_candidates.js]: for newly created sales order there shouldn't be any dunning candidates.

File: cypress/integration/dunning/create_dunning_candidates.js (hunk 3 above)
Comment (TheBestPessimist) [reply cypress/integration/dunning/create_dunning_candidates.js]: fixed

File: cypress/support/commands/action.js (hunk 4 above)
Comment (TheBestPessimist) [reply cypress/support/commands/action.js]: Not yet, as i havent fixed that. will do it in https://github.com/metasfresh/metasfresh-e2e/issues/130 hopefully

Comment (TheBestPessimist) [reply cypress/support/commands/action.js]: NOT fixed!

**Pull request merged**